src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User is null!");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("User's login has to be at least "
+                    + MIN_LOGIN_LENGTH + " characters!");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("User's password  has to be at least "
+                    + MIN_PASSWORD_LENGTH + " characters!");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_AGE) {
+            throw new RegistrationException("User's age should be at least "
+                    + MIN_AGE + " years old!");
+        }
+        return (storageDao.get(user.getLogin()) == null) ? storageDao.add(user) : user;
```

Comment:
Your version is obviously short, but this version of the code is harder to debug and impossible to log, so let's add a few extra lines and make your code easier to read, understand and maintain + let's check if the user already exists and add a test for that.
```suggestion
        if (storageDao.contains(user.getLogin)) {
            throw RegistrationException()
        }
        storageDao.add(user);
        return user;
```

Reply:
I have not implemented this functional only because of "Make sure to not add something that is not described in the task requirements, or tied to something that may change over time like max user age."
And I think "storageDao.contains(user.getLogin)" won't work because our storage stores User objects
;)